lib/ohai/plugins/linux/platform.rb

```diff
@@ -74,12 +86,16 @@ def os_release_file_is_cisco?
       platform get_redhatish_platform(contents)
       platform_version contents.match(/(\d\.\d\.\d)/)[0]
     elsif File.exist?("/etc/redhat-release")
-      if File.exist?('/etc/os-release') && (os_release_info = os_release_file_is_cisco? ) # check if Cisco
-        platform os_release_info['ID']
-        platform_family os_release_info['ID_LIKE']
-        platform_version os_release_info['VERSION'] || ""
+      contents = File.read("/etc/redhat-release").chomp
+      if (os_release_info = os_release_file_is_cisco? ) # check if Cisco
+        if os_release_info['ID'] == 'nexus'
+          platform 'nexus'
+        else
+          platform get_redhatish_platform(contents)
```

Comment:
So this is where I was having a hard time. While I get now how we would end up here, we shouldn't have both this else and the next one. 

If we are on Cisco, and it's nexus, set blah, else were just on red hat, set foo. If we need this many forks, we are going to have a hard time ensuring our test coverage has them all.